src/System.Private.CoreLib/shared/System/Threading/ExecutionContext.cs

```diff
@@ -51,6 +52,37 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
             throw new PlatformNotSupportedException();
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void ResetToDefaults()
+        {
+            Thread currentThread = Thread.CurrentThread;
```

Comment:
ResetToDefaults is called in a loop.  With this aggressively inlined, does this somehow get hoisted as an invariant out of the loop?  Or if not, should it instead be passed in as an argument and grabbed once, since it's a TLS hit?

Reply:
Good idea